### PR TITLE
Router: implement newlines.inInterpolation logic

### LINKED ISSUE #3079: Starting with v3.3.1 scalafmt formats boilerplate code in a weird way
See typelevel/cats#4116

**Configuration**:
```
version=3.3.3
align.openParenCallSite = true
align.openParenDefnSite = true
maxColumn = 120
continuationIndent.defnSite = 2
assumeStandardLibraryStripMargin = true
danglingParentheses.preset = true
rewrite.rules = [AvoidInfix, SortImports, RedundantParens, SortModifiers]
docstrings = JavaDoc
newlines.afterCurlyLambda = preserve
docstrings.style = keep
docstrings.oneline = unfold

runner.dialect = scala213source3
fileOverride {
  "glob:**/*.sbt" {
    runner.dialect = sbt1
  }
  "glob:**/src/{main,test}/scala-2.12/**" {
    runner.dialect = scala212source3
  }
  "glob:**/src/{main,test}/scala-3.x/**" {
    runner.dialect = scala3
  }
}
project.excludeFilters = [
  "scalafix/*"
]

```

## Problem

Scalafmt formats code in boilerplate blocks which is not expected to happen.
Furthermore, the formatted results look pretty ugly: the formatting breaks block margins:
https://github.com/typelevel/cats/pull/4116/files?diff=unified&w=0#diff-dedd06aeff4542c5b99c68ad8ee7fecde2588c7adba68944521cda08e709aa83L153-R154

## Expectation

The boilerplate code blocks should not be formatted by default since the result code can be very different from the source template.

Or at least there should be an option to disable it in configs.

### PR DESCRIPTION
**Pull request by kitbellew**
Fixes #3079.

### CODE REVIEW

File: scalafmt-tests/src/test/resources/default/String.stat [hunk 1]
```diff
@@ -360,3 +360,90 @@ object X {
...
+>>>
+object a {
+  s"foo1 ${
+            bar
+          } foo2 ${baz(bar)} foo3 ${
+                                     qux(baz, bar)
+                                   } foo4"
+  s"""
+    |foo1 ${bar} foo2 ${baz(bar)} foo3 ${qux(baz,
+                                             bar)} foo4
```
Comment (tgodzik) [inline scalafmt-tests/src/test/resources/default/String.stat:447]: Shouldn't there be an additional break here? `foo4` seems to reach maxColumn (51) if I calculated correctly.

**Review by tgodzik — APPROVED** [review]
Looks good, though I am not sure about one example.

File: scalafmt-tests/src/test/resources/default/String.stat (hunk 1 above)
Comment (kitbellew) [reply scalafmt-tests/src/test/resources/default/String.stat:447]: ah... let me check

Comment (kitbellew) [reply scalafmt-tests/src/test/resources/default/String.stat:470]: you were right. added another option, oneline, which is least likely to overflow, and mentioned in the docs the caveat about the others. please take another look.

**Review by tgodzik — APPROVED** [review]
LGTM!

**Comment by satorg** [comment]
@kitbellew thank you for looking into it!

**Pull request merged**